code/game/turfs/simulated/floor.dm

```diff
@@ -82,6 +82,26 @@ GLOBAL_LIST_INIT(icons_to_ignore_at_floor_init, list("damaged1","damaged2","dama
 		if(A.level == 3)
 			return 1
 
+// Checks if the turf is safe to be on
+/turf/simulated/floor/is_safe()
+	if(!air)
+		return FALSE
+	var/datum/gas_mixture/Z = air
+	var/pressure = Z.return_pressure()
+	// Can most things breathe and tolerate the temperature and pressure?
+	if(Z.oxygen < 16 || Z.toxins >= 0.05 || Z.carbon_dioxide >= 10 || Z.sleeping_agent >= 1 || (Z.temperature <= 270) || (Z.temperature >= 360) || (pressure <= 20) || (pressure >= 550))
+		return FALSE
+	return TRUE
+
+// Checks if there is foothold over the turf
+/turf/simulated/floor/proc/find_safeties()
+	var/static/list/safeties_typecache = typecacheof(list(/obj/structure/lattice/catwalk, /obj/structure/stone_tile))
+	var/list/found_safeties = typecache_filter_list(contents, safeties_typecache)
+	for(var/obj/structure/stone_tile/S in found_safeties)
+		if(S.fallen)
+			LAZYREMOVE(found_safeties, S)
```

Comment:
not sure why this is made lazy but....
Just wondering. Don't have to do anything with this since it was already here.